docs: modify backup and restore instructions (#25691)
gravitational/teleport: docs/pages/management/operations/backup-restore.mdx

  - OIDC
- Roles

This command automatically includes all secrets in these resources. You
can also individually retrieve resources with `tctl get` (Example: `tctl get roles`).

When migrating backends, you should back up your Auth Service's
`data_dir/storage` directly.

### Example of backing up and restoring a cluster

```code

> # Run this command on one of the Teleport Auth Service hosts

Reviewer: ```suggestion
# Run this command on one of the Teleport Auth Service hosts.
```
Separate this sentence from the next one

$ tctl --config fresh.yaml get all
# <your state here>
```
<Notice type="danger">

> The `tctl get all` command must be run on a Teleport Auth Service host or existing user records will cause a permission error.

Reviewer 2: Should we add a TODO to remove this when we fix the issue this quarter?

$ tctl --config fresh.yaml get all
# <your state here>
```
<Notice>

> The `tctl get all` requires using on a Teleport Auth Service host 

Reviewer 3: ```suggestion
The `tctl get all` must be run on a Teleport Auth Service host 
```

> <Notice>
> The `tctl get all` must be run on a Teleport Auth Service host 
> or existing User records will cause a permission error.  This is due
> to User secrets only being available from a a direct `tctl` connection
> on a Teleport Auth Service machine. As a workaround, retrieve resources
> individually with the `--with-secrets` parameter except for users. Local user secrets
> including passwords would need to be reset after creating without secrets.

Reviewer: ```suggestion
<Notice type="danger">

The `tctl get all` command must be run on a Teleport Auth Service host or existing user records will cause a permission error.

</Notice>

Teleport user secrets are only available when running `tctl` commands
on a Teleport Auth Service machine. 

As a workaround, retrieve resources
individually with the `--with-secrets` parameter except for users. Local user secrets
including passwords will need to be reset after being created without secrets.
```

- Making the `Notice` smaller and more impactful using the `danger` type. Otherwise the visual qualities of the `Notice` become noise.
- Minor grammar/readability tweaks.

> including passwords would need resetting after creating without secrets.

Reviewer 3: ```suggestion
including passwords would need to be reset after creating without secrets.
```

> All trusted cluster certificate resources should be removed for use

Reviewer: ```suggestion
All Trusted Cluster certificate resources should be removed for use
```
This is capitalized elsewhere in the docs

> in a bootstrap or you will receive an error.  These are the `kind: cert_authority`

Reviewer: What does "removed for use in a bootstrap" mean? Is there another way to phrase this?

> records identifiable with `cluster_name` that match to the leaf cluster names.

Reviewer 3: ```suggestion
records identifiable with `cluster_name` values that match to the leaf cluster names.
```

$ tctl --config fresh.yaml get all
# <your state here>
```
<Notice>

> The `tctl get all` is not fully supported remotely as retrieving user

Reviewer 2: I don't know if I would expect anyone not familiar with Teleport to know what "not fully supported remotely" means..

Author: Please see update @zmb3

resources with secrets is not allowed. As a workaround, retrieve resources
individually. Local user secrets including passwords would need resetting
after creating without secrets.

> All trusted cluster certificate resources should be removed for use

Reviewer 2: Similarly, do we think the reader will know what to do here? This is clear to me, but I suspect it will still be confusing for newcomers.

Author: Please see update @zmb3

# <your state here>
```

> <Notice>

Reviewer: Same comments as the other use of this `Notice`.

Reviewer: @stevenGravy This comment still needs to be addressed. We should edit this `Notice` to match the other `Notice`.

> The `tctl get all` requires using on a Teleport Auth Service host 

Reviewer 3: ```suggestion
The `tctl get all` must be run on a Teleport Auth Service host 
```

or existing User records will cause a permission error.  This is due
to User secrets only being available from a a direct `tctl` connection
on a Teleport Auth Service machine. As a workaround, retrieve resources
individually with the `--with-secrets` parameter except for users. Local user secrets
including passwords would need resetting after creating without secrets.

All trusted cluster certificate resources should be removed for use
in a bootstrap or you will receive an error.  These are the `kind: cert_authority`

> records identifiable with `cluster_name` that match to the leaf cluster names.

Reviewer 3: ```suggestion
records identifiable with `cluster_name` values that match to the leaf cluster names.
```
